### PR TITLE
Week1 

### PR DESCRIPTION
**Pull request by sima-milli**

### CODE REVIEW

**Comment by JawharBairakdar** [comment]
Hi @sima-milli, While I am reviewing your homework as promise, and as you told there is something wrong inside of it, there is two tiny things can be better here ***(Nothing serious but there is a better way always)*** .

Do you remember [this](https://github.com/HackYourFuture/JavaScript2/pull/51/files)?
Without compliments it was the best homework for week1 js2.

what have you done there is:
 - creating the blue-print we can say of your list which you called it *(generateList)*.
 - creating a small tiny pieces of code that make your life easier 🏆.
 - making a very nice clean logic that depends on eachothers *(making a tag with id then using the id to insert an image)* , no need for this here I guess unless you have a very nice Idea of it.

So from here you have the right way and you need someone maybe to lead you in the next step maybe.

Most Important things of What do you have to do?
 - you may create a small tiny pieces of code which make your code on the DRY principle.
 - making a logic of the new **blue-print** which will controle the flow of rendering the page **(your feeling is just right here)** 🙃 .

The bugs you have are:

the error message is showing up with the empty content 👇.
screenshot_2018-08-15 hyf-github
this is because of rendering before the data came from the server, so you have a miss-controle let's say.

And this element right here 👇 may have an effect if you have a larger application that have a deffirent kind of style.
sima-milli-hw
this is fine for now and I am not asking you to change it but think about it.

So what I have seen you may make better, is just to follow the two steps before the images it will give you way better controle over your elements and rendering.

Could you please finish this review before moving to js3 week2?

There is something I referred to, but I didn't tell you about it, and I am expecting you to fegure out it your self 💪 .

**Comment by sima-milli** [comment]
Thanks @JawharBairakdar for your feedback, right now I can't make any changes on my Week1 homework because I have already created a Week2 branch from Week1 
so you may see the changes on Week2 maybe.

**Comment by JawharBairakdar** [comment]
Sounds great, looking forward to see the changes in your homework of the second week 👍

**Comment by sima-milli** [comment]
so you can remove the `needs work` label from this one  :')

**Comment by JawharBairakdar** [comment]
@sima-milli thanks for reply, I am not sure if I can do that at the moment, But I can close it, because keeping `needs work` label is just for a reference for your next Homework which you will Improve your work inside of it.

File: homework/src/index.js [hunk 1]
```diff
@@ -31,17 +32,151 @@
...
+
+        select.addEventListener('change', (e) => {
+          const index = e.target.value;
+          showRepositories(index, data);
+
+          const contributorsInfo = data[index].contributors_url;
+          fetchJSON(contributorsInfo, (err, contributorData) => {
+            ul.innerHTML = '';
+            showContributors(contributorData, ul);
+          });
```
Comment (remarcmij) [inline homework/src/index.js]: This code is a repetition of the code in lines 188-121. You could either create a new function that fetches the contributors and passes them on to `showContributors()` or add the fetching part to the `showContributors()` function (probably then renamed as `fetchAndShowContributors()` like this:

```js
function fetchAndShowContributors(url, ul) {
  fetchJSON(contributorsInfo, (err, contributors) => {
    if (err) {
      //...
   } else {
     // the code that you now have in showContributors
   }
  });
}
```

File: homework/src/index.js [hunk 2]
```diff
@@ -31,17 +32,151 @@
...
+          createAndAppend('option', select, {
+            'html': element.name,
+            'value': i
+          });
+        });
+
+        showRepositories(0, data);
+
+        const contributorsInfo = data[0].contributors_url;
+        fetchJSON(contributorsInfo, (err, contributorData) => {
```
Comment (remarcmij) [inline homework/src/index.js]: Avoid (meaningless) suffixes such as `data`. A better name for `contributorData` would be `contributors`. This plural form suggest that the data is actually an array.

The name `contributorsInfo` is misleading. It is actually a URL so might better be called `contributorsUrl`.

File: homework/src/index.js [hunk 3]
```diff
@@ -31,17 +32,151 @@
...
+      if (data[index].description === null) {
+        descriptionLabel.innerHTML = '';
+        descriptionValue.innerText = '';
+      } else {
+        descriptionLabel.innerText = "Description :";
+        descriptionValue.innerText = data[index].description;
+      }
+      forkValue.innerText = data[index].forks;
+      const updateRepo = new Date(data[index].updated_at);
+      updateValue.innerText = updateRepo.toLocaleString("en-US")
```
Comment (remarcmij) [inline homework/src/index.js]: Why have you chosen `en-US`? It is better to use the computer default (by leaving out the parameter).

File: homework/src/index.js [hunk 4]
```diff
@@ -31,17 +32,151 @@
   }
 
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      'class': 'header'
```
[the remark this answers is not in the document]
Comment (sima-milli) [reply homework/src/index.js]: got it 👍

File: homework/src/index.js [hunk 5]
```diff
@@ -31,17 +32,151 @@
...
+        descriptionValue.innerText = data[index].description;
+      }
+      forkValue.innerText = data[index].forks;
+      const updateRepo = new Date(data[index].updated_at);
+      updateValue.innerText = updateRepo.toLocaleString("en-US")
+    }
+
+    function showContributors(contributorData, ul) {
+      contributorData.forEach(contributor => {
+        const link = createAndAppend('a', ul, {
```
[the remark this answers is not in the document]
Comment (sima-milli) [reply homework/src/index.js]: Aha, ok 👍

File: homework/src/index.js [hunk 6]
```diff
@@ -31,17 +32,151 @@
...
+          html: err.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {
+          ignorePunctuation: true
+        }));
+
+        data.forEach((element, i) => {
```
[the remark this answers is not in the document]
Comment (sima-milli) [reply homework/src/index.js]: done

File: homework/src/index.js (hunk 2 above)
Comment (sima-milli) [reply homework/src/index.js]: done 👍

**Pull request closed without merging**